Cat3FGL.py

```diff
@@ -1,371 +1,455 @@
+
+        # elif self.tbdata['SpectrumType']=="PLExpCutoff":
+        # I gather "PLExpCutoff" and "PLSuperExpCutoff" distinguished via parameters:
+        
+        else:  
+            K=self.tbdata['Flux_Density'][i]
+            Gamma=self.tbdata['Spectral_Index'][i]
+            E0=self.tbdata['Pivot_Energy'][i]
+            Ec=self.tbdata['Cutoff'][i]
+            b=self.tbdata['Exp_Index'][i]
+            self.spectral_model = lambda E: K*(E/E0)**(-Gamma)*np.exp((E0/Ec)**b - (E/Ec)**b)
+
+        return n
+
+
+
+    def get_all_fields(self):
+        """Return all info about selected object"""
+
+        if self._selected():
+            return self.tbdata[self.index]
+
+
+
+    def get_field(self, field):
+        """
+        Return a given field - field should be passed as a string.
+        Most useful:'Source_Name', 'SpectrumType', 'Spectral_Index', 'Powerlaw_Index',
+        'Pivot_Energy', 'Flux_Density', 'Flux300_1000', 
+        see: https://fermi.gsfc.nasa.gov/ssc/data/access/lat/4yr_catalog/
+        """
+
+        if self._selected():
+            return self.tbdata[field][self.index]
+
+
+    def calc_PL_int_flux(self, E_low_MeV, E_high_MeV):
+        """
+        Return the calculated integral flux between two energies based on powerlaw model
+        if the spectrum type is a PowerLaw
+        """ 
+        
+        if not self._selected():
+            return -1
+
+        if not self.tbdata['SpectrumType'][self.index]=='PowerLaw':
+            return -2
+        
+
+        E_pivot=self.tbdata['Pivot_Energy'][self.index]
+        gamma=self.tbdata['Powerlaw_Index'][self.index]
+        C=self.tbdata['Flux_Density'][self.index]
+       
+        resultPL=C/(1-gamma)*(E_high_MeV**(1-gamma) - E_low_MeV**(1-gamma)) / E_pivot**(-gamma)
+        resultPLunits = resultPL*u.Unit('cm-2 s-1') #adding proper units
+        return resultPLunits
+        
+
+
+    def calc_int_flux(self, E_low_MeV, E_high_MeV):
+        """
+        Return the calculated integral flux between two by numerically integrating the best-fit model
+        """ 
+
+        if not self._selected():
+            return -1
+
+        
+        import scipy.integrate as integrate
+        result=integrate.quad(self.spectral_model,E_low_MeV, E_high_MeV)
+        
+        return result[0]
+        
+
+
+    def get_field_names(self):
+        """Return all of the field names"""
+
+        return self.tbdata.names
+
+
+    def get_Flux300_1000(self):
+        """Return all of the field names"""
+
+        if self._selected():
+            return self.tbdata['Flux300_1000'][self.index]
+
+
+    def get_SpectrumType(self):
+        """Return all of the field names"""
+
+        if self._selected():
+            return self.tbdata['SpectrumType'][self.index]
+        
+##################################EBL########################################
+
+
+
+    def calc_int_absorbed_flux_gammapy(self,E_low, E_high,model,redshift):
+        
+        #input in MeV

```

Comment:
Is input in MeV or TeV? - code below says TeV? I suggest naming the variables to include units, e.g. E_low_MeV or E_low_TeV are adding function help (comments inside """ """) which explains which to use.